src/AutoRest.CSharp/MgmtTest/Generation/ResourceTestWriter.cs

```diff
@@ -0,0 +1,214 @@
+using AutoRest.CSharp.Mgmt.Decorator;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using AutoRest.CSharp.Mgmt.Models;
+using AutoRest.CSharp.Utilities;
+
+namespace AutoRest.CSharp.MgmtTest.Generation
+{
+    /// <summary>
+    /// Code writer for resource test.
+    /// </summary>
+    internal class ResourceTestWriter : MgmtBaseTestWriter
+    {
+        protected Resource _resource;
+        protected string TestNamespace => _resource.Type.Namespace + ".Tests.Mock";
+        protected override string TypeNameOfThis => _resource.Type.Name + "MockTests";
+
+        protected string TestBaseName => $"MockTestBase";
+
+        protected ResourceCollectionTestWriter? _collectionTestWriter;
+
+        public ResourceTestWriter(CodeWriter writer, Resource resource, BuildContext<MgmtOutputLibrary> context): base(writer, resource, context)
+        {
+            _resource = resource;
+            if (!resource.IsSingleton)
+            {
+                _collectionTestWriter = new ResourceCollectionTestWriter(writer, resource.ResourceCollection!, context);
+                if (CanCreateResourceFromExample(context, resource.ResourceCollection!))
+                {
+                    _collectionTestWriter.EnsureCollectionInitiateParameters();
+                }
+            }
+        }
+
+        public void WriteCollectionTest()
+        {
+            WriteUsings(_writer);
+            _writer.UseNamespace("System.Net");
+            _writer.UseNamespace("System.Collections.Generic");
+            _writer.UseNamespace("Azure.Core.TestFramework");
+            _writer.UseNamespace("Azure.ResourceManager.TestFramework");
+            _writer.UseNamespace("Azure.ResourceManager.Resources");
+            _writer.UseNamespace("Azure.ResourceManager.Resources.Models");
+
+            using (_writer.Namespace(TestNamespace))
+            {
+                _writer.WriteXmlDocumentationSummary($"Test for {_resource.ResourceName}");
+                _writer.Append($"public partial class {TypeNameOfThis:D} : ");
+                _writer.Line($"{TestBaseName}");
+                using (_writer.Scope())
+                {
+                    WriteTesterCtors();
+                    if (!_resource.IsSingleton)
+                    {
+                        _collectionTestWriter!.WriteCreateCollectionMethod(true);
+                    }
+                    WriteCreateResourceMethod(true);
+                    WriteMethodTestIfExist(_resource);
+
+                    foreach (var childResource in This.ChildResources)
+                    {
+                        _writer.Line();
+                        if (childResource.IsSingleton) {
+                            WriteMethodTestIfExist(childResource);
+                        }
+                    }
+                }
+            }
+        }
+
+        protected void WriteMethodTestIfExist(Resource resource) {
```

Comment:
We need to format the code
```suggestion
        protected void WriteMethodTestIfExist(Resource resource)
        {
```